docs/translations/translations-lazy-loading.md.md

```diff
@@ -0,0 +1,138 @@
+# Lazy-load translations
+
+Some of our translations are lazy-loaded. To lazy-load them, the translation chunks needs to be registered. This takes place in `/src/BlockTypes/AbstractBlock.php`:
+
+```php
+/**
+ * Injects Chunk Translations into the page so translations work for lazy loaded components.
+ *
+ * The chunk names are defined when creating lazy loaded components using webpackChunkName.
+ *
+ * @param string[] $chunks Array of chunk names.
+ */
+protected function register_chunk_translations( $chunks ) {
+  foreach ( $chunks as $chunk ) {
+    $handle = 'wc-blocks-' . $chunk . '-chunk';
+    $this->asset_api->register_script( $handle, $this->asset_api->get_block_asset_build_path( $chunk ), [], true );
+    wp_add_inline_script(
+      $this->get_block_type_script( 'handle' ),
+      wp_scripts()->print_translations( $handle, false ),
+      'before'
+    );
+    wp_deregister_script( $handle );
+  }
+}
+```
+
+## Lazy-loaded translations of the Cart block
+
+The translations of the inner blocks of the Cart block are loaded in this function in `src/BlockTypes/Cart.php`:
+
+```php
+/**
+ * Register script and style assets for the block type before it is registered.
+ *
+ * This registers the scripts; it does not enqueue them.
+ */
+protected function register_block_type_assets() {
+  parent::register_block_type_assets();
+  $blocks = [
+    'cart-blocks/express-payment--checkout-blocks/express-payment--checkout-blocks/payment',
+    'cart-blocks/line-items',
+    'cart-blocks/order-summary',
+    'cart-blocks/order-summary--checkout-blocks/billing-address--checkout-blocks/shipping-address',
+    'cart-blocks/checkout-button',
+    'cart-blocks/express-payment',
+  ];
+  $chunks = preg_filter( '/$/', '-frontend', $blocks );
+  $this->register_chunk_translations( $chunks );
+}
+```
+
+## Lazy-loaded translations of the Checkout block
```

Comment:
```suggestion
## Lazy-loaded translations of the Checkout Block
```

Reply:
I thought we were using sentence case for our blocks:

- Cart block
- Checkout block
- etc.

I cannot find the corresponding documentation though, and I do see that we're using both versions in our codebase. I'd leave the parts unchanged until we have a final consent in this area.